spec/integration/indirector/report/yaml.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+require 'puppet/transaction/report'
+require 'puppet/indirector/report/yaml'
+require 'puppet/indirector/indirector_testing/report/yaml'
+
+describe Puppet::Transaction::Report::Yaml do
+  describe '#save' do
+    let(:model) { Puppet::IndirectorTesting }
+    let(:indirection) { model.indirection }
+    let(:instance) { model.new('report') }
+    let(:request)  { indirection.request(:report, 'example', instance) }
```

Comment:
For save requests, the `indirection.request` method is called with `:save` as the first parameter, and request key as the second parameter, where the request key is the report's hostname.

However, this kind of setup is fragile. When working on the rest termini changes, I wrote the rest_spec tests to call the public indirector API. For example, configure the indirection to use a specific terminus:

https://github.com/puppetlabs/puppet/blob/master/spec/unit/indirector/report/rest_spec.rb#L15-L17

Then call the Indirection#save method, passing in the report:

https://github.com/puppetlabs/puppet/blob/master/spec/unit/indirector/report/rest_spec.rb#L42